database/migrations/2024_08_01_193038_remove_and_rebuild_wiki_lifecycle_events.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+use App\Jobs\UpdateWikiSiteStatsJob;
+use App\WikiLifecycleEvents;
+use Illuminate\Database\Migrations\Migration;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        WikiLifecycleEvents::all()->map->delete();
+        (new UpdateWikiSiteStatsJob())->handle();
```

Comment:
Does this run synchronously and waits until all jobs for all wikis are finished or will it just dispatch a set of jobs into the queue and then move on? Asking because I would think calling `handle` on a job is a wee bit of an anti pattern, but maybe we just need it.

Reply:
> Does this run synchronously and waits until all jobs for all wikis are finished

It does this exactly; runs them synchronously and if they fail the migration will fail. You think it would be better to just dispatch it to the queue?

Reply:
I would think that the migration runs in a transaction, and this means we have a transaction open for a very long time. But maybe that's ok.

Reply:
yeah, I would guess it be open for best part of a minute+. I've made the job dispatchable and instead dispatched it.

To paste from my local setup this of 50 Wikis from ~10s to 32ms to run the migration